src/main/groovy/life/qbic/datamodel/dtos/business/ProductId.groovy

```diff
@@ -23,27 +23,86 @@ class ProductId {
     /**
      * Identifying number used in conjunction with the type
      */
-    private final String identifier
+    private final long uniqueId
+
+    /**
+     * A builder for ProductId instances.
+     */
+    static class Builder {
+        private String productType
+        private long uniqueId
+
+        /**
+         *
+         * @param productType
+         * @param identifier the unique id - will be interpreted as unsigned long
+         */
+        Builder(String productType, String identifier) {
```

Comment:
I think the Typ is also fixed for the different products (or should be) like DS for data storage.

Reply:
Shall I change it? Do we haven an enum already? I have the impression that we use the distinction rather often. And I would suggest to implement that in the classes to get their product type string